cvat/apps/annotation/validation.py

```diff
@@ -0,0 +1,214 @@
+import io
+from itertools import groupby
+
+from cvat.apps.annotation.ddln_spotter_importer import label_by_class_id
+from cvat.apps.annotation.structures import BoundingBox
+
+
+def validate(sequences):
+    reporter = ValidationReporter()
+    for seq in sequences:
+        reporter.sequence = seq.name
+        class_id_by_track_id = {}
+        previous_frame = None
+        sequence_track_ids = set()
+
+        for frame in seq.frames:
+            reporter.frame = frame.name
+            reporter.count_frame(seq.name)
+
+            frame_track_ids = set()
+            duplicated_track_ids = set()
+            for bbox_index, bbox in enumerate(frame.bboxes, start=1):
+                reporter.bbox_index = bbox_index
+
+                if bbox.track_id in frame_track_ids:
+                    duplicated_track_ids.add(bbox.track_id)
+                frame_track_ids.add(bbox.track_id)
+                sequence_track_ids.add(bbox.track_id)
+
+                _validate_coordinates(bbox, reporter)
+                _validate_attributes(bbox, reporter)
+                _validate_class_immutability(bbox, class_id_by_track_id, reporter)
+                _validate_position_change(bbox, previous_frame, jump_threshold=2, reporter=reporter)
+
+            reporter.bbox_index = None
+            for track_id in duplicated_track_ids:
+                reporter.report_track_id_duplication(track_id)
+            previous_frame = frame
+
+        reporter.frame = None
+        _validate_track_id_are_consecutive(sequence_track_ids, reporter)
+    return reporter
+
+
+def _validate_attributes(bbox, reporter):
+    try:
+        int(bbox.track_id)
+    except (TypeError, ValueError):
+        reporter.report_invalid_track_id_value(bbox.track_id)
+
+    if bbox.class_id not in label_by_class_id:
+        reporter.report_invalid_class_id_value(bbox.class_id)
+
+
+def _validate_class_immutability(bbox, class_id_by_track_id, reporter):
```

Comment:
NB: many of these validations can be hardcoded in annotation UI, so that annotators cannot make these mistakes